Change kotlin to java

charafau opened the pull request:
Moved kotlin code to java for avoiding strange errors

charafau committed:
- Moved kotlin code to java for avoiding strange errors
- Added settings user in example

charafau committed:
changed map to argument method

jaumard reviewed (changes requested):
Thanks for this !!! Here is my first review, I'll take time to test it on my side too ;)

charafau committed:
changed ifs to switch statement
fixed regression - added putting stack trace to flutter exception
added else statements to if
make FlutterException to extends RuntimeException

jaumard reviewed (approved):
I get a chance to test it :) all good on my side! Thanks again for this !

willconno commented:
@charafau Can you elaborate more on the 'strange errors' you found?
